server/api_server/tools_routes.py

```
@@ -0,0 +1,685 @@
+        conn = get_temp_db_connection()
+        conn.row_factory = sqlite3.Row
+        cur = conn.cursor()
+        try:
+            # Try to find device by IP (devLastIP)
+            cur.execute("SELECT devMac FROM Devices WHERE devLastIP = ?", (ip,))
+            row = cur.fetchone()
+            if row and row['devMac']:
+                mac = row['devMac']
+            else:
+                return jsonify({"error": f"Could not resolve MAC for IP {ip}"}), 404
+        except Exception as e:
```

Ruff (0.14.6)

Do not catch blind exception: `Exception`

(BLE001)